File: output/llama_4_maverick/avatar/algo_based_translation/java/python/failed_translation/codeforces_189_A.py

def main():
    n = int(input().split()[0])
    a, b, c = map(int, input().split())
    maxn = 4010
    arr = [a, b, c]
    dp = [float('-inf')] * maxn
    dp[0] = 0
    for i in range(len(arr)):
        for j in range(arr[i], maxn):
            dp[j] = max(dp[j], dp[j - arr[i]] + 1)
    print(dp[n])

if __name__ == "__main__":
    main()

File: output/deepseek_chat/avatar/algo_based_translation/python/java/codeforces_189_A.java

import java.util.*;

public class codeforces_189_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int[] arr = new int[input.length];
        for (int i = 0; i < input.length; i++) {
            arr[i] = Integer.parseInt(input[i]);
        }
        int length = arr[0];
        int[] pieces = Arrays.copyOfRange(arr, 1, 4);
        Arrays.sort(pieces);
        if (length % pieces[0] == 0) {
            System.out.println(length / pieces[0]);
            return;
        }
        int ans = 0;
        if (pieces[0] == pieces[1]) {
            for (int i = 0; i <= length / pieces[2]; i++) {
                int check = length - i * pieces[2];
                if (check >= 0 && check % pieces[0] == 0) {
                    int k = check / pieces[0];
                    ans = Math.max(ans, k + i);
                }
            }
        } else {
            for (int i = 0; i <= length / pieces[2]; i++) {
                for (int j = 0; j <= length / pieces[1]; j++) {
                    int check = length - i * pieces[2] - j * pieces[1];
                    if (check >= 0 && check % pieces[0] == 0) {
                        int k = check / pieces[0];
                        ans = Math.max(ans, k + i + j);
                    }
                }
            }
        }
        System.out.println(ans);
    }
}

